src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/IMaturedBlockDepositsBuilder.cs

```diff
@@ -0,0 +1,11 @@
+﻿using NBitcoin;
+using Stratis.Bitcoin.Primitives;
+using Stratis.FederatedPeg.Features.FederationGateway.Models;
+
+namespace Stratis.FederatedPeg.Features.FederationGateway.Interfaces
+{
+    public interface IMaturedBlockDepositsProcessor
+    {
+        MaturedBlockDepositsModel ExtractMaturedBlockDeposits(ChainedHeaderBlock latestPublishedBlock);
```

Comment:
I think it is better to return the interfaced type (IMaturedBlockDeposits) rather than the concrete one